### PR TITLE
Add type definition for parse5

### PR DESCRIPTION
**Pull request by inikulin**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

-----

Starting from `5.0.0` parse5 will no longer provide its own type definitions and will be split into separate packages  (see https://github.com/inikulin/parse5/issues/235).

### CODE REVIEW

File: types/parse5-sax-parser/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+/// <reference types="node" />
+
+import * as stream from "stream";
+import * as parse5 from "parse5";
+
+export interface StartTagToken {
+    /**
+     * Tag name.
+     */
+    name: string;
```
Comment (amiller-gh) [inline types/parse5-sax-parser/index.d.ts]: I believe this is `tagName`

Comment (DanielRosenwasser) [reply types/parse5-sax-parser/index.d.ts]: @inikulin can you verify?

Comment (inikulin) [reply types/parse5-sax-parser/index.d.ts]: Fixed

**Comment by DanielRosenwasser** [comment]
Just as a heads up, your commits don't seem to be associated with your GitHub account. While this isn't technically a problem, you might care if you want more appropriate attribution. You can either make sure your GitHub account is associated with the email address you're using for your commits, or rebase and amend your commits to fix the author name and email.

If you don't care about any of this stuff, I can always just pull in the changes.

**Comment by inikulin** [comment]
@DanielRosenwasser Thanks for pointing this out, I always forget to switch accounts. Fixed now.

**Pull request merged**